h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstSpearman.java

```diff
@@ -0,0 +1,180 @@
+package water.rapids.ast.prims.advmath;
+
+import water.DKV;
+import water.MRTask;
+import water.Scope;
+import water.Value;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.Env;
+import water.rapids.Val;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.rapids.vals.ValNum;
+import water.util.FrameUtils;
+
+public class AstSpearman extends AstPrimitive<AstSpearman> {
+  @Override
+  public int nargs() {
+    return 1 + 3; // Frame ID and ID of two numerical vectors to calculate SCC on.
+  }
+
+  @Override
+  public String[] args() {
+    return new String[]{"frame", "first_column", "second_column"};
+  }
+
+  @Override
+  public Val apply(Env env, Env.StackHelp stk, AstRoot[] asts) {
+    final Value value = DKV.get(asts[1].str());
+    if (!value.isFrame()) {
+      throw new IllegalArgumentException(String.format("The given key '%s' is not a frame.", asts[1]));
+    }
+
+    final Frame originalUnsortedFrame = value.get(Frame.class);
+    final int vecIdX = originalUnsortedFrame.find(asts[2].exec(env).getStr());
+    final int vecIdY = originalUnsortedFrame.find(asts[3].exec(env).getStr());
+
+    Scope.enter();
```

Comment:
Scope always needs to be used with try-finally

Reply:
Done. Moved the whole ranking logic to a separate method wrapped in try-finally block, so the scope is ensured to be exitted.